Manage a nic used for north/south traffic

gnuoy opened the pull request:
If a nic is supplied for use for north/south traffic then wire it into br-ex. Also manage removing it if /reset is called.

Commit:
Manage a nic used for north/south traffick

If a nic is supplied for use for north/south traffic then wire it
into br-ex. Also manage removing it if /reset is called.

gnuoy commented:
Corresponding microstack PR https://github.com/openstack-snaps/snap-microstack/pull/55

javacruft commented on openstack_hypervisor/api.py:
```diff
@@ -141,6 +142,14 @@ async def update_logging(config: model.LoggingConfig):
+        external_bridge = snap.config.get("network.external-bridge")
+        hooks.del_interface_from_bridge(external_bridge, gateway_nic)
+    except UnknownConfigKey:
+        pass
```
will and UNSET key throw a UnknownConfigKey error?  if not then we should probably validate that the nic option is actually set

javacruft commented:
One inline comment and a general observation; we have the opportunity to record which interface was configured by the snap in OVS as part of the additional metadata we can store in OVS.

This would allow us to detect if the configuration option was changed - the old interface can be removed and the new one added thus avoiding someone creating a spanning tree loop in their network or other such horribleness.

This is implemented in the ovn charm layer for the existing machine charms:

https://github.com/openstack-charmers/charm-layer-ovn/blob/master/lib/charms/ovn_charm.py#L1246

Commit:
Use ovsdb to store microstack ext-nic info

Tag the microstack managed external nic so it can be correctly
removed on reset.

javacruft commented on openstack_hypervisor/hooks.py:
```diff
@@ -499,6 +500,105 @@ def _configure_ovn_base(snap: Snap) -> None:
+    for p in external_ports:
+        if p != external_nic:
+            logging.debug(
+                f"Removing additional extranal ports {external_ports} from {external_bridge}"
```
typo: extranal -> external

javacruft replied:
and should this be {p} rather than the full list?

Commit:
fix error in log statement
